xds/internal/xdsclient/client_new.go

```diff
@@ -54,24 +57,48 @@ const NameForServer = "#server"
 // only when all references are released, and it is safe for the caller to
 // invoke this close function multiple times.
 func New(name string) (XDSClient, func(), error) {
-	return newRefCounted(name, defaultWatchExpiryTimeout, defaultIdleAuthorityDeleteTimeout, backoff.DefaultExponential.Backoff)
+	return newRefCounted(name, defaultWatchExpiryTimeout, defaultIdleChannelExpiryTimeout, backoff.DefaultExponential.Backoff)
 }
 
 // newClientImpl returns a new xdsClient with the given config.
-func newClientImpl(config *bootstrap.Config, watchExpiryTimeout time.Duration, idleAuthorityDeleteTimeout time.Duration, streamBackoff func(int) time.Duration) (*clientImpl, error) {
+func newClientImpl(config *bootstrap.Config, watchExpiryTimeout, idleChannelExpiryTimeout time.Duration, streamBackoff func(int) time.Duration) (*clientImpl, error) {
 	ctx, cancel := context.WithCancel(context.Background())
 	c := &clientImpl{
 		done:               grpcsync.NewEvent(),
+		authorities:        make(map[string]*authority),
 		config:             config,
 		watchExpiryTimeout: watchExpiryTimeout,
 		backoff:            streamBackoff,
 		serializer:         grpcsync.NewCallbackSerializer(ctx),
 		serializerClose:    cancel,
+		transportBuilder:   &grpctransport.Builder{},
 		resourceTypes:      newResourceTypeRegistry(),
-		authorities:        make(map[string]*authority),
-		idleAuthorities:    cache.NewTimeoutCache(idleAuthorityDeleteTimeout),
+		xdsActiveChannels:  make(map[string]*channelState),
+		xdsIdleChannels:    cache.NewTimeoutCache(idleChannelExpiryTimeout),
 	}
 
+	for name, cfg := range config.Authorities() {
+		// If server configus are specified in the authorities map, use that. Else,
```

Comment:
just for my understanding, `config.XDSServers()` will be there only in case of fallback servers?